core/src/test/java/org/elasticsearch/monitor/fs/FsProbeTests.java

```diff
@@ -91,6 +92,33 @@ public void testFsInfo() throws IOException {
         }
     }
 
+    public void testFsInfoOverflow() throws Exception {
+        try (NodeEnvironment env = newNodeEnvironment()) {
+            FsProbe probe = new FsProbe(Settings.EMPTY, env);
+
+            FsInfo stats = probe.stats(null, null);
+            assertNotNull(stats);
+
+            FsInfo.Path total = stats.getTotal();
+            assertNotNull(total);
+            assertThat(total.total, greaterThan(0L));
+
+            // While not overflowing, keep adding
+            while ((total.total + total.total) > 0) {
```

Comment:
If `total` is pathological (like one byte), this will take forever. Can you just add invocations of `randomNonNegativeLongs()`?